### PR TITLE
Move base costing options parsing into a separate function

### PR DESCRIPTION
**Pull request by genadz**
# Issue

https://github.com/valhalla/valhalla/issues/3102

## Tasklist

 - [ ] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Update the docs with any new request parameters or changes to behavior described
 - [ ] Update the [changelog](CHANGELOG.md)
 - [ ] If you made changes to the lua files, update the [taginfo](taginfo.json) too.

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: src/sif/dynamiccost.cc
Comment (genadz) [inline src/sif/dynamiccost.cc:297]: moved to `ParseBaseCostOptions`

File: src/sif/pedestriancost.cc
Comment (genadz) [inline src/sif/pedestriancost.cc:474]: we don't parse `tool_booth_cost` option for pedestrians, so it doesn't make sense to use it here

File: valhalla/sif/dynamiccost.h
Comment (mandeepsandhu) [inline valhalla/sif/dynamiccost.h:1031]: s/gate_base_costs/get_base_costs

**Comment by kevinkreiser** [comment]
@genadz sorry i havent looked at this yet. i will schedule it for over the weekend.

**Comment by genadz** [comment]
> @genadz sorry i havent looked at this yet. i will schedule it for over the weekend.

thank you!

**Review by mandeepsandhu — APPROVED** [review]
Thanks for the refactor @genadz! This will make working with costings much easier! 

This lgtm. We should wait for @kevinkreiser 's feedback before merging it in.

**Comment by kevinkreiser** [comment]
no need to wait for me! judging by the magnitude of the change and the importance id just suggest that we RAD it if we havent already just to be sure. code motion copy pasting type stuff is easy to make a typo imho

**Review by kevinkreiser — APPROVED** [review]
Looks good just a RAD of each costing would be good. Although maybe we can do without that with a more mechanical test. Like a loop over all costings over all costing args? I know we have the online costing tests but I forget if they are exhastive

**Comment by merkispavel** [comment]
I RAD tested the changes - no failures

**Comment by genadz** [comment]
> I RAD tested the changes - no failures 🟢

Thank you! Could you merge it?

**Pull request merged**
